fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/EnumMapper.kt

```diff
@@ -0,0 +1,12 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.middleware
+
+/**
+ * Converts a string to an enum value, ignoring case. If the string does not match any of the
+ * enum values, the default value is returned.
+ */
+inline fun <reified T : Enum<T>> String.asEnumOrDefault(defaultValue: T? = null): T? =
+    enumValues<T>().firstOrNull { it.name.equals(this, ignoreCase = true) } ?: defaultValue
```

Comment:
are these enumMapper changes related to recommendations? should they be in this PR

Reply:
Yes, since this will be used in the recommendations mapping logic (next PR) along with the 3 places it's used currently, the idea was to split this bug into 2 PRs(commits) in effort to make it easier for reviews, so the second part won't have these changes of extracting the enum mapper and tests.